### PR TITLE
check api resources before install proxy

### PR DESCRIPTION
**Pull request by YitongFeng**
Related issue #1153

### CODE REVIEW

File: pai-management/k8sPaiLibrary/maintainlib/deploy.py [hunk 1]
```diff
@@ -197,6 +197,9 @@ def run(self):
         kubectl_install_instance = kubectl_install.kubectl_install(self.cluster_config)
         kubectl_install_instance.run()
 
+        # check the registerd api resources
+        common.execute_shell("kubectl api-resources", "kubectl command failed!")
```
Comment (ydye) [inline pai-management/k8sPaiLibrary/maintainlib/deploy.py]: Maybe you should retry util successful

Comment (YitongFeng) [reply pai-management/k8sPaiLibrary/maintainlib/deploy.py]: Good point

File: pai-management/k8sPaiLibrary/maintainlib/deploy.py [hunk 2]
```diff
@@ -203,6 +203,12 @@ def run(self):
         kubectl_install_instance = kubectl_install.kubectl_install(self.cluster_config)
         kubectl_install_instance.run()
 
+        # check the registerd api resources
+        common.execute_shell_retry("kubectl api-resources", "kubectl command failed!", 5)
+
+        # create kube-proxy until daemonset resource is registerd
+        common.execute_shell_retry("kubectl api-resources | grep -q daemonsets", "Controller manager hasn't create daemonset object!", 5)
```
Comment (hao1939) [inline pai-management/k8sPaiLibrary/maintainlib/deploy.py]: 1. Will `kubectl api-resources | grep -q daemonsets`  return none zero when daemonsets not ready?
2. Does the grep -q case-sensitive?

Comment (YitongFeng) [reply pai-management/k8sPaiLibrary/maintainlib/deploy.py]: yes, only when it grep nothing it return 0, I have test

Comment (YitongFeng) [reply pai-management/k8sPaiLibrary/maintainlib/deploy.py]: kubectl api-resources results are all lowercase, the "grep" is case-sensitive

**Comment by hao1939** [comment]
Fixes #1153

**Pull request merged**